libcloud/compute/drivers/verizon.py

```diff
@@ -0,0 +1,890 @@
+        return self._to_node(result).state
+
+    def ex_get_node(self, node=None):
+        """
+        Returns a node
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :return: State of the node
+        :rtype: :class: `Node`
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        result = self.connection.request(node.id).object
+        return self._to_node(result)
+
+    def ex_poweroff_node(self, node=None):
+        """
+        Power off a node
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :return: State of the node
+        :rtype: :class: `Node`
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        href = None
+        for i in node.extra['Actions']:
+            if i is not None:
+                if i['name'] == 'power:powerOff':
+                    href = i['href']
+        headers = {'Content-Length': 0}
+        return self.connection.request(href,headers=headers,method='POST').object
+
+    def ex_remove_node(self, node=None):
+        """
+        Deletes a node
+        :keyword node: Node object
+        :type    node: :class:`Node` (mandatory)
+
+        :return: State of the node
+        :rtype: :class: `Node`
+        """
+        if node is None:
+            raise VerizonException('Parameter node is required')
+        headers = {'Content-Length': 0}
+        return self.connection.request(node.id,headers=headers,method='DELETE').object
+
+    def _to_node(self, node):
+        public_ips = []
+        private_ips = []
+        state = 'UNKNOWN'
+
+        if node['IpAddresses']['AssignedIpAddresses']['Networks'] is not None:
+            for i in node['IpAddresses']['AssignedIpAddresses']['Networks']:
+                for a in i['IpAddresses']:
+                    public_ips.append(a)
+
+        if 'ip' in node:
+            public_ips.append(node['ip'])
+
+        if node['PoweredOn'] is True:
+            state = 'ON'
+        if node['PoweredOn'] is False:
+            state = 'OFF'
+        if node['Status'] == '3':
+            state = 'IN_PROGRESS'
+        if node['CustomizationPending'] is True:
+            state = 'IN_PROGRESS'
+
+        return Node(node['href'], node['name'], NODE_STATE_MAP[state], public_ips, private_ips, driver=self, extra=node)
+
+    def _to_image(self, image):
+        return NodeImage(id=image['href'], name=image['name'], driver=self)
+
+    def _libcloud_layout_check(self):
+        """
+        Check to see if a libcloud row and column exist. If not, create
+        """
+        env = self.ex_get_default_env()
+        url = API_ROOT + '/layout/environments/' + env
+        rows = self.connection.request(url).object['Rows']
+        row = None
+        grp = None
+        for i in rows:
+            if i['name'] == 'libcloud':
```

Comment:
Is it possible to refactor this in multiple methods to avoid that many levels of nesting which makes code hard to read?